xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -182,11 +182,23 @@ class PredicateAudioFilter
     PREDICATE_RETURN(lh.type_index == currentAudioStream
                      , rh.type_index == currentAudioStream);
 
-    if (!StringUtils::EqualsNoCase(CServiceBroker::GetSettings().GetString(CSettings::SETTING_LOCALE_AUDIOLANGUAGE), "original"))
+    PREDICATE_RETURN(lh.flags & StreamFlags::FLAG_FORCED,
```

Comment:
please drop this, FORCED is for subs, not audio. why do you want to force a user to hear a particular audio stream?

Reply:
@fernetmenta:

1.) It's a proper solution for your russion guy who sideloads a track.
2.) We most probably transport it already for mkv streams, at least its documented there:
> FlagForced | 3 | [55][AA] | mand. | - | 0-1 | 0 | u | * | * | * | * | * | Set if that track MUST be active during playback. There can be many forced track for a kind (audio, video or subs), the player should select the one which language matches the user preference or the default + forced track. Overlay MAY happen between a forced and non-forced track of the same kind. (1 bit)
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --

Seems not to be only for subs - does it hurt?

Reply:
1.) It's a proper solution for your russion guy who sideloads a track.
2.) We most probably transport it already for mkv streams, at least its documented there:

FlagForced | 3 | [55][AA] | mand. | - | 0-1 | 0 | u | * | * | * | * | * | Set if that track MUST be active during playback. There can be many forced track for a kind (audio, video or subs), the player should select the one which language matches the user preference or the default + forced track. Overlay MAY happen between a forced and non-forced track of the same kind. (1 bit)
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --

Seems not to be only for subs - does it hurt?

Reply:
> the player should select the one which language matches the user preference or the default + forced track

this is not what your code does. it is what PredicateSubtitleFilter does

> does it hurt?

I don't see the benefit. Apart from that we support multi-demuxer and multi source. Adopting Matroshka spec (single source) does not make any sense to me.